spec/components/previews/provider_interface/change_offer_preview.rb

```diff
@@ -0,0 +1,151 @@
+module ProviderInterface
+  class ChangeOfferPreview < ViewComponent::Preview
+    def existing_offer_start_at_1_provider(provider_interface_change_offer_form: nil)
```

Comment:
The first 3 methods here repeat the same code varying only on the `initial_step` method argument. That suggests something common could be called.

Reply:
Most code in this file has been added to provide a scaffold so that clicking through the various sub-components works (more or less). However, this is not a controller, so I cannot rely on any `:before_action` magic or equivalent. This means I would need to extract the common code into a method and call this from each scenario, maybe saving 1-2 lines in each case but sacrificing readability.

The reason we don't have only two cases (existing offer, new offer) is that subcomponent titles and change links at the end of the process are conditional to `entry`, therefore starting the flow at different points is significant (mostly for the existing offer flow).

Reply:
For me it's not the number of lines of code we'd save so much as repeating the exact same lines several times (with the exception of the value passed to `initial_step`), maintaining methods like this will become a headache. 
I'm perhaps not fully understanding how these methods are called as I can't seem to find where they are called.

Reply:
These methods are only called in the context of component previews, see https://apply-for-te-1995-refac-01ltkk.herokuapp.com/rails/view_components/provider_interface/change_offer 

The previews themselves are used for easy visual inspection and testing components in isolation. They are similar to tests, in that they also document things.

The fact these methods are only ever called for as part of development and QA is why I think some repetition may be ok here.